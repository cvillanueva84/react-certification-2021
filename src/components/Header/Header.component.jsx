import React, { useState } from 'react';
import styled from 'styled-components';
import Toggle from '../Toggle';

const Container = styled.div`
  background-color: ${(props) => props.theme.header};
  color: ${(props) => props.theme.text};
  display: flex;
  position: fixed;
  top: 0;
  height: 9vh;
  max-height: 55px;
  width: 100vw;
  justify-content: space-around;
  align-items: center;
  border-bottom: solid 1px ${(props) => props.theme.secondary};
`;

const Logo = styled.h1`
  font-size: 1.75rem;
  user-select: none;
  cursor: pointer;
  span {
    padding: 0px 5px;
    background-color: red;
    color: white;
    border-radius: 7.5px;
    position: relative;
    &::after {
      content: 'TV';
      color: #696969;
      position: absolute;
      font-size: 0.65rem;
      top: 0;
      right: -1rem;
    }
  }
`;

const SearchBar = styled.div`
  background-color: ${(props) => props.theme.accent};
  height: 5vh;
  max-height: 30px;
  width: 45%;
  display: flex;
  align-items: center;
  font-size: 1rem;
  color: ${(props) => props.theme.text};
  &:hover {
    filter: brightness(${(props) => props.theme.filter});
  }
  svg {
    cursor: pointer;
    height: 90%;
    margin: 0px 10px;
    fill: ${(props) => (props.searchTerm ? props.theme.text : '#696969')};
  }
  input {
    border: none;
    background-color: inherit;
    font-size: inherit;
    height: 100%;
    width: 90%;
    color: ${(props) => props.theme.text};
    &::placeholder {
      color: #696969;
    }
    &:focus,
    &:active {
      outline: none;
    }
  }
`;

const ProfilePicture = styled.div`
  position: relative;
  width: 38px;
  height: 38px;
  border-radius: 50px;
  cursor: pointer;
  background-color: ${(props) => props.theme.text};
  border: solid 2px ${(props) => props.theme.text};
  overflow: hidden;
  &::after {
    position: absolute;
    content: '';
    width: 15px;
    height: 15px;
    border-radius: 50px;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: ${(props) => props.theme.header};
  }
  &::before {
    position: absolute;
    content: '';
    width: 30px;
    height: 15px;
    border-radius: 50px;
    left: 50%;
    transform: translate(-50%, 175%);
    background-color: ${(props) => props.theme.header};
  }
`;

function SearchInput() {
  const [searchTerm, setSearchTerm] = useState('');
  return (
    <SearchBar searchTerm={searchTerm}>
      <svg
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
        aria-hidden="true"
      >
        <path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z" />
      </svg>
      <input
        type="text"
        placeholder="Search"
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
      />
    </SearchBar>
  );
}

function Header({ themeToggler }) {
  return (
    <Container>
      <Logo>
        Wize<span>line</span>
      </Logo>
      <SearchInput />
      <Toggle action={themeToggler} />
      <ProfilePicture />
    </Container>
  );
}

export default Header;
