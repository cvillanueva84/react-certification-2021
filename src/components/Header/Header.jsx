import React, { useState, useContext } from 'react';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import Switch from '@material-ui/core/Switch';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { AppContext } from '../../providers/App/App.provider';
import { primary, dark } from '../../style/theme';
import {
  Container,
  LeftHeader,
  RightHeader,
  SearchDiv,
  DivSearchIcon,
  DivInput,
  ButtonAccount,
  DivAvatar,
} from './Header.styled';

function Header({ children, enterPressed }) {
  const [checked, setChecked] = useState(false);
  const { state, dispatch } = useContext(AppContext);
  const toggleChecked = () => {
    setChecked((prev) => !prev);
    if (checked) {
      dispatch({ type: 'SET_THEME', payload: { theme: primary } });
    } else {
      dispatch({ type: 'SET_THEME', payload: { theme: dark } });
    }
  };
  return (
    <Container theme={state.theme}>
      <LeftHeader>
        <MenuIcon />
        <SearchDiv type="submit" onSubmit={enterPressed}>
          <DivInput>
            <DivSearchIcon>
              <SearchIcon />
            </DivSearchIcon>
            {children}
          </DivInput>
        </SearchDiv>
      </LeftHeader>
      <RightHeader>
        <div>
          <FormGroup>
            <FormControlLabel
              control={<Switch checked={checked} onChange={toggleChecked} />}
              label="Dark Mode"
            />
          </FormGroup>
        </div>
        <ButtonAccount>
          <DivAvatar>
            <AccountCircleIcon style={{ fontSize: 40 }} />
          </DivAvatar>
        </ButtonAccount>
      </RightHeader>
    </Container>
  );
}

export default Header;
