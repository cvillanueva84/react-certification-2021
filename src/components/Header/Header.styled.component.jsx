/* eslint-disable import/no-extraneous-dependencies */
import styled from 'styled-components';
import { Menu } from '@styled-icons/boxicons-regular/Menu';

export const HeaderContainer = styled.header`
  background-color: #1c5476;
  color: white;
  min-height: 64px;
  width: 100%;
  display: flex;
  align-self: flex-between;
  padding: 0 24px;
`;

export const LeftHeaderContainer = styled.div`
  display: flex;
  justify-content: start;
  width: 100%;
`;

export const BurgerMenuContainer = styled.div`
  /* padding: 12px 20px; */
  display: flex;
  justify-content: center;
  min-width: 48px;
`;

export const BurgerMenu = styled(Menu)`
  color: ${(props) => props.color};
  margin-right: 16px;
  padding: 12px 0;
`;
