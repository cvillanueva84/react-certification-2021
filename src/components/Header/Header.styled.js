import styled from 'styled-components';

const HeaderWrapper = styled.div`
  background-color: transparent;
  height: 60px;
  display: flex;
  width: 100%;
  align-items: center;
  padding: 10px 20px;
  justify-content: space-between;
  border-bottom: 1px solid #dcdcdc;
`;

const FlexCentered = styled.div`
  display: flex;
  align-items: center;
`;

const Input = styled.input`
  padding: 10px;
  width: 400px;
  border: 1px solid #8e8e8e;
  height: 41px;
`;

const Button = styled.button`
  padding: 10px;
  background: none;
  border: 1px solid #8e8e8e;
  width: 70px;
  background-color: #8e8e8e;
  height: 41px;
`;

export { Button, Input, FlexCentered, HeaderWrapper };
