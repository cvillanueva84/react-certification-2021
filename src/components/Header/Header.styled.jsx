import styled from 'styled-components';

export const Container = styled.header`
  color: white;
  width: 100%;
  height: 64px;
  display: flex;
  background-color: ${(props) => props.theme['background-header']};
  justify-content: space-between;
  align-items: center;
`;
export const LeftHeader = styled.div`
  display: flex;
  align-items: center;
  margin-left: 30px;
`;
export const RightHeader = styled.div`
  display: flex;
  align-items: center;
  margin-right: 30px;
  @media (max-width: 768px) {
    display: none;
  }
`;

export const SearchDiv = styled.form`
  width: auto;
  margin-left: 24px;
  position: relative;
  margin-right: 16px;
  border-radius: 4px;
  background-color: rgba(255, 255, 255, 0.2);
`;
export const DivSearchIcon = styled.button`
  display: flex;
  color: white;
  height: 100%;
  padding: 0px 16px;
  position: absolute;
  align-items: center;
  pointer-events: none;
  justify-content: center;
  background-color: transparent;
  z-index: 1;
  border: none;
`;

export const DivInput = styled.div`
  display: inline-flex;
  position: relative;
  align-items: center;
`;

export const SwitchBtn = styled.span`
  width: 58px;
  height: 38px;
  display: inline-flex;
  padding: 12px;
  z-index: 0;
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  flex-shrink: 0;
  vertical-align: middle;
`;

export const IconButton = styled.span`
  top: 0;
  left: 0;
  color: #fafafa;
  z-index: 1;
  position: absolute;
  transition: left 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
    transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 9px;
  overflow: visible;
  font-size: 1.5rem;
  text-align: center;
  border-radius: 50%;
  border: 0;
  margin: 0;
  display: inline-flex;
  outline: 0;
  align-items: center;
  user-select: none;
  vertical-align: middle;
  justify-content: center;
  text-decoration: none;
  background-color: transparent;
`;

export const LabelIconButton = styled.span`
  width: 100%;
  display: flex;
  color: white;
  font-size: 1.5rem;
  text-align: center;
  align-items: inherit;
  justify-content: inherit;
`;

export const Input = styled.input`
  left: -100%;
  width: 300%;
  top: 0;
  height: 100%;
  z-index: 1;
  position: absolute;
  padding: 0;
  opacity: 0;
  cursor: inherit;
  margin: 0;
`;

export const Thumb = styled.span`
  color: white;
  width: 20px;
  height: 20px;
  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%),
    0px 1px 3px 0px rgb(0 0 0 / 12%);
  border-radius: 50%;
  background-color: currentColor;
`;

export const TouchRipple = styled.span`
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 0;
  overflow: hidden;
  position: absolute;
  border-radius: inherit;
  pointer-events: none;
`;

export const TrackSwitch = styled.span`
  width: 100%;
  height: 100%;
  opacity: 0.38;
  z-index: -1;
  transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
    background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 7px;
  background-color: #000;
`;

export const ButtonAccount = styled.button`
  color: inherit;
  height: 100%;
  padding: 12px;
  margin-left: 12px;
  align-items: center;
  cursor: pointer;
  outline: 0;
  background-color: transparent;
  border: 0;
`;

export const DivAvatar = styled.div`
  width: 40px;
  height: 40px;
  overflow: hidden;
  align-items: center;
`;
