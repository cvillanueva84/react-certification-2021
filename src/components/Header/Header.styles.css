/***************MENU TOGGLE****************/
.header-icon span {
  cursor: pointer;
}

/***************SEARCH BAR****************/
.search-bar {
  align-items: center;
  display: flex;
}

.search-bar input,
.search-bar button {
  border: none;
  height: 2.5em;
}

.search-bar input {
  background-color: #c2c3ff;
  border-radius: 4px 0 0 4px;
  color: #191a5e;
  width: 100%;
}

.search-bar button {
  background-color: #191a5e;
  border-radius: 0 4px 4px 0;
  cursor: pointer;
}

/***************PROFILE MENU****************/
.input-icon {
  background-color: #c2c3ff;
  border-radius: 50%;
  cursor: pointer;
  height: 2em;
  width: 2em;
}
