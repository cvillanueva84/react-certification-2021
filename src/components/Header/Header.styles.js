import styled from "styled-components";
import hamburguer from '../../img/menu.png';
import profile from '../../img/user.png';

export const HeaderBody = styled.div`
    display: flex;
    align-items: center;
    text-align: center;
`;

export const HeaderContainer = styled.div`
    display: flex;
    justify-content: space-between;
    width: 100vw;
    min-height: 60px;
    background: #1C5476;
    padding: 15px 20px;
    color: #fff;
    display: flex;
    box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%), 0px 4px 5px 0px rgb(0 0 0 / 14%), 0px 1px 10px 0px rgb(0 0 0 / 12%);
`;

export const Hamburguer = styled.span`
    background-image: url(${hamburguer});
    width: 30px;
    height: 30px;
    background-size: contain;
    background-repeat: no-repeat;
`;

export const Search = styled.input`
    padding: 10px;
    margin: 5px 0px;
    background: white;
    border: none;
    border-radius: 3px;
    height: 30px;
    margin-left: 30px;
    background-color: #3E6D8A;
    color: white;
    
    ::placeholder {
        color: white;
    }

    @media (max-width: 485px) {
        margin-left: 15px;
    }
    @media (max-width: 768px) {
        
    }
`;

export const ToggleBox = styled.input`    
    border-radius: 15px;
    width: 42px;
    height: 26px;
    margin-right: 30px;

    @media (max-width: 485px) {
        margin-right: 15px;
    }
    @media (max-width: 768px) {
        
    }
`;

export const Profile = styled.span`
    background-image: url(${profile});
    width: 30px;
    height: 30px;
    background-size: contain;
    background-repeat: no-repeat;
`;