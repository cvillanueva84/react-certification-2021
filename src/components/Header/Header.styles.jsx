import styled from 'styled-components';

const HeaderBody = styled.div`
  border-bottom: solid 1px #f1f1f1;
  display: flex;
  justify-content: space-between;
  padding: 0.75rem;
  @media (max-width: 1000px) {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(25rem, 1fr));
    gap: 0.5rem;
  }
`;
const HeaderLeft = styled.div`
  display: flex;
  gap: 2rem;
`;
const HeaderItems = styled.div`
  display: flex;
  gap: 2rem;
  @media (max-width: 1000px) {
    display: none;
  }
`;
const Search = styled.input`
  padding: 16px 16px 16px 60px;
  border: 1px solid;
  border-radius: 3px;
  box-shadow: inset 0 -8px 1px rgba(248, 10, 10, 0.03);
  font-weight: 300;
  font-size: 15px;
  width: 400px;
  height: 30px;
  :hover {
    background-color: #f8f7f7;
  }
  ::placeholder,
  ::-webkit-input-placeholder {
    color: #dbd8d8;
    font-weight: 300;
    font-size: 1.1rem;
  }
`;
const Button = styled.button`
  font-size: 1em;
  font-weight: 600;
  padding: 0.1rem 1rem;
  text-align: center;
  border-radius: 50%;
  background-image: url(${(props) => props.avatarUrl});
  background-position: center;
  background-size: 60px 50px;
  background-repeat: no-repeat;
  :hover {
    border: 2px solid black;
  }
`;
const CheckBoxWrapper = styled.div`
  position: relative;
  top: 10px;
`;
const CheckBoxLabel = styled.label`
  position: absolute;
  left: 0;
  width: 42px;
  border-radius: 15px;
  background: #bebebe;
  &::after {
    content: '';
    display: block;
    border-radius: 50%;
    width: 18px;
    height: 18px;
    margin: 3px;
    background: #ffffff;
    box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.2);
    transition: 0.2s;
  }
`;
const CheckBox = styled.input`
  opacity: 0;
  z-index: 1;
  border-radius: 15px;
  width: 42px;
  height: 26px;
  &:checked + ${CheckBoxLabel} {
    background: red;
    &::after {
      content: '';
      display: flex;
      border-radius: 50%;
      width: 18px;
      height: 18px;
      margin-left: 21px;
      transition: 0.2s;
    }
  }
`;

export {
  HeaderBody,
  HeaderLeft,
  HeaderItems,
  Search,
  Button,
  CheckBoxWrapper,
  CheckBoxLabel,
  CheckBox,
};
