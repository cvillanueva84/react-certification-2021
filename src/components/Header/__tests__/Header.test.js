import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { Header } from '../Header.component';
import { AppContext } from '../../../context/appContext';

describe('Testing Header component', () => {
  const contextValue = {
    dispatch: jest.fn(),
    state: {
      search: 'wizeline',
      showVideoDetails: false,
    },
  };

  beforeEach(() => {
    render(
      <AppContext.Provider value={contextValue}>
        <Header />
      </AppContext.Provider>
    );
  });

  it('should take a snapshop', () => {
    expect(screen).toMatchSnapshot();
  });

  it('should be text "Dark mode"', () => {
    expect(screen.queryByText(/Dark mode/)).toBeInTheDocument();
  });

  it('should change the input', () => {
    const input = screen.getByPlaceholderText('Search');
    fireEvent.change(input, { target: { value: 'hi' } });

    expect(input.value).toBe('hi');
  });

  it('should change the input', () => {
    const input = screen.getByPlaceholderText('Search');
    fireEvent.change(input, { target: { value: '' } });

    expect(contextValue.dispatch).not.toHaveBeenCalled();
  });
});
