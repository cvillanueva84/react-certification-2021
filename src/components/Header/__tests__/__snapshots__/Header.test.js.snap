// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component should take a snapshop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-eCApnc hRyBgW"
      >
        <ul
          class="sc-jSFjdj fkiwxZ"
        >
          <li
            class="sc-gKAaRy hKcdfX"
          >
            <button
              class="sc-ksluID hikfVZ"
            >
              <i
                class="fas fa-bars fa-2x"
              />
            </button>
          </li>
          <li
            class="sc-gKAaRy hKcdfX"
          >
            <form
              class="sc-fujyAs jgUeTi"
            >
              <button
                class="sc-pNWdM eAAEIR"
                type="submit"
              >
                <i
                  class="fas fa-search fa-lg"
                />
              </button>
              <input
                class="sc-jrsJWt fksNle"
                placeholder="Search"
                type="text"
                value="wizeline"
              />
            </form>
          </li>
          <div
            class="sc-iCoGMd itDCGY"
          />
          <div
            class="sc-kEqXSa jjDVFg"
            on="md"
          >
            <li
              class="sc-gKAaRy hKcdfX"
            >
              <div
                class="sc-bdnxRM hFrJuC"
              >
                <p
                  class="sc-hKFxyN SUEDR"
                >
                  Dark mode
                </p>
                <label
                  class="sc-gtsrHT iezsSu"
                >
                  <input
                    type="checkbox"
                  />
                  <span
                    class="sc-dlnjwi Itfri"
                  />
                </label>
              </div>
            </li>
          </div>
          <div
            class="sc-kEqXSa jjDVFg"
            on="md"
          >
            <li
              class="sc-gKAaRy hKcdfX"
            >
              <button
                class="sc-ksluID hikfVZ"
              >
                <i
                  class="fas fa-user-circle fa-2x"
                />
              </button>
            </li>
          </div>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-eCApnc hRyBgW"
    >
      <ul
        class="sc-jSFjdj fkiwxZ"
      >
        <li
          class="sc-gKAaRy hKcdfX"
        >
          <button
            class="sc-ksluID hikfVZ"
          >
            <i
              class="fas fa-bars fa-2x"
            />
          </button>
        </li>
        <li
          class="sc-gKAaRy hKcdfX"
        >
          <form
            class="sc-fujyAs jgUeTi"
          >
            <button
              class="sc-pNWdM eAAEIR"
              type="submit"
            >
              <i
                class="fas fa-search fa-lg"
              />
            </button>
            <input
              class="sc-jrsJWt fksNle"
              placeholder="Search"
              type="text"
              value="wizeline"
            />
          </form>
        </li>
        <div
          class="sc-iCoGMd itDCGY"
        />
        <div
          class="sc-kEqXSa jjDVFg"
          on="md"
        >
          <li
            class="sc-gKAaRy hKcdfX"
          >
            <div
              class="sc-bdnxRM hFrJuC"
            >
              <p
                class="sc-hKFxyN SUEDR"
              >
                Dark mode
              </p>
              <label
                class="sc-gtsrHT iezsSu"
              >
                <input
                  type="checkbox"
                />
                <span
                  class="sc-dlnjwi Itfri"
                />
              </label>
            </div>
          </li>
        </div>
        <div
          class="sc-kEqXSa jjDVFg"
          on="md"
        >
          <li
            class="sc-gKAaRy hKcdfX"
          >
            <button
              class="sc-ksluID hikfVZ"
            >
              <i
                class="fas fa-user-circle fa-2x"
              />
            </button>
          </li>
        </div>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
